Fixes this build error:

In file included from /usr/lib64/ocaml/caml/io.h:28,
                 from mlmpfr_stubs.h:323,
                 from mlmpfr_stubs.c:17:
/usr/lib64/ocaml/caml/platform.h: In function ‘caml_plat_latch_is_released’:
/usr/lib64/ocaml/caml/platform.h:222:10: error: implicit declaration of function ‘atomic_load_acquire’ [-Wimplicit-function-declaration]
  222 |   return atomic_load_acquire(&latch->value) == Latch_released;
      |          ^~~~~~~~~~~~~~~~~~~
/usr/lib64/ocaml/caml/platform.h: In function ‘caml_plat_latch_set’:
/usr/lib64/ocaml/caml/platform.h:230:3: error: implicit declaration of function ‘atomic_store_release’; did you mean ‘atomic_store_explicit’? [-Wimplicit-function-declaration]
  230 |   atomic_store_release(&latch->value, Latch_unreleased);
      |   ^~~~~~~~~~~~~~~~~~~~
      |   atomic_store_explicit

--- mlmpfr-mlmpfr.4.2.1/src/mlmpfr_stubs.h.orig	2023-12-10 13:02:38.000000000 -0700
+++ mlmpfr-mlmpfr.4.2.1/src/mlmpfr_stubs.h	2024-12-09 14:31:18.501078409 -0700
@@ -17,12 +17,16 @@
 #ifndef MLMPFR_STUBS_H
 #define MLMPFR_STUBS_H
 
+#include <unistd.h>
+// FIXME Try a way to not include CAML_INTERNALS things
+#define CAML_INTERNALS
 #include <caml/alloc.h>
 #include <caml/callback.h>
 #include <caml/custom.h>
 #include <caml/fail.h>
 #include <caml/memory.h>
 #include <caml/mlvalues.h>
+#undef CAML_INTERNALS
 #include <limits.h>
 #include <stdio.h>
 /* Prototypes for MPFR functions with FILE * parameters are provided only if
